desc/objectives/objective_funs.py

```diff
@@ -786,13 +800,44 @@ def __init__(
         weight=1,
         normalize=True,
         normalize_target=True,
+        loss_function=None,
+        where_apply_loss="compute_unscaled",
         name=None,
     ):
+        assert where_apply_loss in ["compute_unscaled", "compute_scaled"], (
+            "where_apply_loss must be 'compute_unscaled' or"
+            f"'compute_scaled', got {where_apply_loss}"
+        )
+
+        if loss_function is None:
+            loss_function = lambda x: x
+        loss_function_on_unscaled = (
+            loss_function if where_apply_loss == "compute_unscaled" else lambda x: x
+        )
+        loss_function_on_scaled = (
+            loss_function if where_apply_loss == "compute_scaled" else lambda x: x
+        )
+
         assert np.all(np.asarray(weight) > 0)
         assert normalize in {True, False}
         assert normalize_target in {True, False}
         assert (bounds is None) or (isinstance(bounds, tuple) and len(bounds) == 2)
         assert (bounds is None) or (target is None), "Cannot use both bounds and target"
+        assert callable(loss_function), "Loss function must be callable!"
+        # function should be able to handle input of this type and shape
+        try:
+            _test_output_loss = loss_function(jnp.ones((2,)))
```

Comment:
any reason the user couldn't create a loss function that specialized on the shape of the expected input? Would this break that?

Reply:
No, this was just to test that the function accepts arrays. There is probably a better way by having it accept dim_f size array and ensuring no error is raised